File: frontend/src/types/activities.ts
export interface SchemaProperty {
    type: 'string' | 'number' | 'boolean' | 'object' | 'array';
    title?: string;
    default?: any;
}

export interface InputSchema {
    type: "object";
    required?: string[];
    properties: Record<string, SchemaProperty>
}

export interface ActivityDefinition {
    id: number;
    name: string;
    activity_type: string;
    node_type: string;
    input_schema: InputSchema | null;  // 可能為 null（如 Standup, Sitdown）
    created_at: string;
    updated_at: string;
}

File: backend/internal/repository/models/activities.go
package models

import (
	"time"

	"gorm.io/datatypes"
)

type ActivityDefinition struct {
	ID           int            `json:"id" gorm:"primaryKey autoIncrement"`
	Name         string         `json:"name" gorm:"not null"`
	ActivityType string         `json:"activity_type" gorm:"uniqueIndex;not null"`
	NodeType     string         `json:"node_type" gorm:"default:'default'"`
	InputSchema  datatypes.JSON `json:"input_schema" gorm:"type:json"`
	CreatedAt    time.Time      `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt    time.Time      `json:"updated_at" gorm:"autoUpdateTime"`
}
